### PR TITLE
curl to get assets

### PR DESCRIPTION
**Pull request by gautam-shantanu**
This script demonstrates the usage of netbackup REST API for listing the assets

### CODE REVIEW

File: snippets/curl/get_nb_assets.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                                ;;
+                esac
+                shift 2
+        done
+
+        if [ -z "$master_server" ] || [ -z "$username" ] || [ -z "$password" ] || [ -z "$domainname" ] || [ -z "$domaintype" ]; then
+                showhelp
+        fi
+
+        if [ "${domaintype^^}" = "WINDOWS" ] || [ "${domaintype^^}" = "NT" ]; then
```
Comment (rkennedy) [inline snippets/curl/get_nb_assets.sh:65]: Beware. The `^^` parameter expansion is a Bash extension. It's not Posix, and might not work on all versions of sh.

File: snippets/curl/get_nb_assets.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
+!/bin/sh
+
+#####################n#####################################################
+
+# This script demonstrates the usage of netbackup REST API for listing
```
Comment (rkennedy) [inline snippets/curl/get_nb_assets.sh]: Capitalize _NetBackup_, please.

```suggestion
# This script demonstrates the usage of NetBackup REST API for listing
```

File: snippets/curl/get_nb_assets.sh [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+master_server=$master_server.$domainname
+basepath="https://$master_server:$port/netbackup"
+content_header='content-type:application/json'
+
+##############login#############
+
+uri="$basepath/login"
+echo $uri
+
+data=$(jq --arg name $username --arg pass $password --arg dname $domainname --arg dtype $domaintype \
```
Comment (rkennedy) [inline snippets/curl/get_nb_assets.sh]: Quote shell parameters on the command line. I don't see `dname` or `dtype` used, but they're required parameters.

```suggestion
data=$(jq --arg name "$username" --arg pass "$password" --arg dname "$domainname" --arg dtype "$domaintype" \
```

File: snippets/curl/get_nb_assets.sh [hunk 4]
```diff
@@ -0,0 +1,116 @@
+!/bin/sh
```
Comment (rkennedy) [inline snippets/curl/get_nb_assets.sh]: The shebang is missing the "she" part.

```suggestion
#!/bin/sh
```

File: snippets/curl/get_nb_assets.sh [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+#####################n#####################################################
```
Comment (rkennedy) [inline snippets/curl/get_nb_assets.sh]: Is the _n_ significant?

```suggestion
###########################################################################
```

File: snippets/curl/get_nb_assets.sh [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+password=""
+domainname=""
+domaintype=""
+
+showHelp()
+{
+        echo ""
+        echo "Invalid command parameters"
+        echo "Usage:"
+        echo "./get_nb_vmservers.sh -nbmaster <master_server> -username <username> -password <password> -domainname <dname> -domaintype <unixpwd/nt>"
```
Comment (rkennedy) [inline snippets/curl/get_nb_assets.sh]: Command doesn't match the file name.

```suggestion
        echo "./get_nb_assets.sh -nbmaster <master_server> -username <username> -password <password> -domainname <dname> -domaintype <unixpwd/nt>"
```

**Review by rkennedy — DISMISSED** [review]
Could use a little improvements, but could go in as-is, too.

**Review by gautam-shantanu — COMMENTED** [review]
review comments addressed

**Review by rkennedy — APPROVED** [review]

**Pull request merged**